code/modules/clothing/clothing.dm

```diff
@@ -542,6 +562,32 @@
 	icon_state = "[icon_state]_opentoe"
 	item_state = "[item_state]_opentoe"
 
+/obj/item/clothing/shoes/AltClick(mob/user)
+	if(user.stat || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED) || !Adjacent(user))
+		return
+
+	if(!knife_slot)
+		return
+	if(!hidden_blade)
+		to_chat(user, "<span class='warning'>There's nothing in your [name]!</span>")
+		return
+
+	if(isobj(user.get_active_hand()) && isobj(user.get_inactive_hand()))
+		to_chat(user, "<span class='warning'>You need an empty hand to pull [hidden_blade]!</span>")
+		return
+	else
+		user.visible_message("<span class='notice'>[user] pulls [hidden_blade] from their [name]!</span>", \
+			"<span class='notice'>You draw [hidden_blade] from your [name]!</span>")
+		user.put_in_hands(hidden_blade)
+		hidden_blade.add_fingerprint(user)
+		hidden_blade = null
+
+/obj/item/clothing/shoes/examine(mob/user)
+	. = ..()
+
+	if(knife_slot)
+	. += "<span class='info'>You can <b>Alt-Click</b> [src] to remove a stored knife. Use the knife on the shoes to place one in [src]</span>"
```

Comment:
```suggestion
	. += "<span class='info'>You can <b>Alt-Click</b> [src] to remove a stored knife. Use the knife on the shoes to place one in [src].</span>"
```

Reply:
Just caught this one, last second change momento.